docs/guides/styled-components.md

```diff
@@ -0,0 +1,116 @@
+# Styled components
+
+Ah yes, begone are the old days of css files, getting a scss or sass compiler working,... Welcome styled components!
+The benefits are:
+
+- Dynamic css, pass js variables to your css.
+- Unique classnames, no magic dangling css class that mess up everything.
+
+## There's a plugin for that
+
+Get the plugin and install it according to it's [documentation](https://github.com/nozzle/react-static/tree/master/packages/react-static-plugin-styled-components).
+
+## Let's convert the basic example to use styled components
+
+More detailed information on how to use styled components can be found [here](https://www.styled-components.com/docs/api).
+
+Add the styled components core library first:
+
+```
+  yarn add styled-components
+```
+
+Remove the `app.css` file.
+
+Open `app.js` and remove the `import './app.css'` so we're not using that css anymore.
+
+### Global styling
+
+Import the global style function at the top.
+
+```
```

Comment:
Add a language annotation to this code block, so that the highlighting works:

````markdown
```javascript
import { createGlobalStyle } from "styled-components";
```
````